lib/ask/runtime/session.ex

```diff
@@ -424,6 +425,28 @@ defmodule Ask.Runtime.Session do
     session |> handle_step_answer(step_answer, current_mode)
   end
 
+  # If the respondent has answered at least `min_relevant_steps` relevant steps
+  # and the reply doesn't defines already a disposition
+  # then, 'interim partial' disposition is returned in reply
+  defp relevant_interim_partial_step({:ok, flow, %{disposition: nil} = reply} = step_answer, %{disposition: "started"} = respondent) do
+    new_step_answer = if Flow.interim_partial_by_relevant_steps?(flow) do # Filtered here to avoid fetching the responses unnecessarily
```

Comment:
Besides checking if the step is relevant or not we are missing:
* to only set `interim partial` if the respondent is in `started`
* optimization, if the number of answers is less than `min_relevant_steps`, then there is nothing else to check.

What happen if the first question answered (which I guess is setting the respondent to `started`) it is also expected to set the respondent as `interim partial` ?

Reply:
1. the function definition uses pattern matching to only apply for `disposition: started` respondents
``` elixir
defp relevant_interim_partial_step({:ok, flow, step_answer}, %{disposition: "started"} = respondent)
```
2. Thought about it at first, but when I coded it seemed to me like a premature optimization, so I removed it. The only thing that we would be saving would be the 
``` elixir
|> Enum.count(&Flow.relevant_response?(flow, &1))
```
that's traversing a collection in-memory and comparing it with in-memory values, since we would still have to go fetch the responses to the db

3. _if the first question answered it is also expected to set the respondent as interim partial_: ✅ works. Added a test to validate it